src/selectors/entities/channels.js

```diff
@@ -923,3 +923,28 @@ export const getSortedDirectChannelWithUnreadsIds = createIdsSelector(
     filterChannels,
 );
 
+// Filters post IDs by the given condition.
+// The condition function receives as parameters the associated channel object and the post object.
+export const filterPostIds = (condition) => {
+    if (typeof condition !== 'function') {
+        throw new TypeError(`${condition} is not a function`);
+    }
+    return createSelector(
+        getAllChannels,
+        getAllPosts,
+        (state, postIds) => postIds,
+        (channels, posts, postIds) => {
+            const ids = [];
+            postIds.forEach((postId) => {
```

Comment:
Why a forEach and not a `for ... in` loop, or a `.filter`?

Reply:
No particular reason... are you thinking what I'm thinking, namely:
a) `for ... in` would be faster, and
b) `filter` would avoid an extra variable declaration?